core/designsystem/src/commonMain/kotlin/io/github/droidkaigi/confsched2023/designsystem/theme/Theme.kt

```diff
@@ -72,6 +73,43 @@ private val DarkColors = darkColorScheme(
     scrim = md_theme_dark_scrim,
 )
 
+sealed class HallColorScheme {
+    abstract val hallA: Color
+    abstract val hallB: Color
+    abstract val hallC: Color
+    abstract val hallD: Color
+    abstract val hallE: Color
+    abstract val hallText: Color
+    abstract val hallTextWhenWithoutSpeakers: Color
+
+    data class Light(
+        override val hallA: Color = md_theme_light_room_hall_a,
+        override val hallB: Color = md_theme_light_room_hall_b,
+        override val hallC: Color = md_theme_light_room_hall_c,
+        override val hallD: Color = md_theme_light_room_hall_d,
+        override val hallE: Color = md_theme_light_room_hall_e,
+        override val hallText: Color = md_theme_light_room_hall_text,
+        override val hallTextWhenWithoutSpeakers: Color = md_theme_light_onSurfaceVariant,
+    ) : HallColorScheme()
+
+    data class Dark(
+        override val hallA: Color = md_theme_dark_room_hall_a,
+        override val hallB: Color = md_theme_dark_room_hall_b,
+        override val hallC: Color = md_theme_dark_room_hall_c,
+        override val hallD: Color = md_theme_dark_room_hall_d,
+        override val hallE: Color = md_theme_dark_room_hall_e,
+        override val hallText: Color = md_theme_dark_room_hall_text,
+        override val hallTextWhenWithoutSpeakers: Color = md_theme_dark_onSurfaceVariant,
+    ) : HallColorScheme()
+}
+
+@Composable
+fun HallColors() = if (isSystemInDarkTheme()) {
```

Comment:
Looks great! But maybe we have to use lower case for the Composable function which returns a value.